app/Models/EventLog.php

```diff
@@ -41,21 +40,59 @@ class EventLog extends Model
     ];
 
     /**
-    * The attributes that use some Models\Traits with mutator settings automation
-    *
-    * @var array
-    */
+     * The attributes that use some Models\Traits with mutator settings automation.
+     *
+     * @var array
+     */
     protected $traitAutomate = [
-      'cid' => 'BinHex',
-      'tid' => 'BinHex',
-      'pay_to' => 'AccountAddress',
-      'ip' => 'BinHex',
-      'our_context' => 'JsonValue',
-      'their_context' => 'JsonValue',
-      'user_id' => 'BinHex',
-      'our_userdata' => 'JsonValue',
-      'their_userdata' => 'JsonValue',
-      'event_value' => 'Money',
-      'paid_amount' => 'Money',
+        'cid' => 'BinHex',
+        'tid' => 'BinHex',
+        'pay_to' => 'AccountAddress',
+        'ip' => 'BinHex',
+        'our_context' => 'JsonValue',
+        'their_context' => 'JsonValue',
+        'user_id' => 'BinHex',
+        'our_userdata' => 'JsonValue',
+        'their_userdata' => 'JsonValue',
+        'event_value' => 'Money',
+        'paid_amount' => 'Money',
     ];
+
+    public function getAdpayJson()
+    {
+        return [
+            'event_id' => (string) $this->id,
+            'event_type' => $this->event_type,
+            'event_value' => $this->event_value,
+            'banner_id' => (string) $this->banner_id,
+            'our_keywords' => Utils::flattenKeywords($this->getOurKeywords()),
+            'their_keywords' => Utils::flattenKeywords($this->getTheirKeywords()),
+            'timestamp' => $this->updated_at,
+            'user_id' => $this->user_id,
+            'advertiser_id' => 1, // TODO: chat with Jacek
+            'human_score' => $this->human_score,
+        ];
+    }
+
+    public function getOurKeywords()
+    {
+        return array_merge(
+            (array) $this->our_context,
+            [
+                'user' => $this->our_userdata,
+            ]
+          );
+    }
+
+    public function getTheirKeywords()
+    {
+        return array_merge(
+            (array) $this->their_context,
+            [
+                'user' => $this->their_userdata,
+            ]
+        );
+
+        return $data;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Review the data-flow - use of uninitialized value. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=php%3AS836)